bson/bson_test.go

```diff
@@ -177,13 +178,22 @@ func TestMapCodec(t *testing.T) {
 			{"true", bsonoptions.MapCodec().SetEncodeKeysWithStringer(true), "bar"},
 			{"false", bsonoptions.MapCodec().SetEncodeKeysWithStringer(false), "foo"},
 		}
+		buf := new(bytes.Buffer)
+		enc := new(Encoder)
```

Comment:
It's technically safe to share the buffer and `Encoder` in this case, but it creates an opportunity for unexpected interaction between test cases. We should initialize these values in the test case funcs.